kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanBuilderImpl.java

```diff
@@ -89,7 +91,11 @@ public Scan build() {
   }
 
   @Override
-  public PaginatedScan buildPaginated(long pageSize, Optional<Row> pageToken) {
-    throw new UnsupportedOperationException("not implemented");
+  public PaginatedScan buildPaginated(long pageSize, Optional<Row> pageTokenRowOpt) {
+    Scan baseScan = this.build();
```

Comment:
You can actually just change `public Scan build() {` to be `public ScanImpl build() {` -- recall, it is totally valid for a child implementation (ScanBuilderImpl) to return a child type (ScanImpl) instead of the parent type (Scan) :)

and then there is no need to `checkArgument`